### ISSUE TITLE
generate the none caption result

### ISSUE THREAD

**Issue by AlvinAi96** [elliottd/satyrid#5]
Hi ellittd,

I try to implement your code, but while training, the sample results are all none which means there is nothing be generated. Also, while validation during the training process, the costs of valid dataset and test dataset are all shown as -0.000. I am sure my input is right but I don't know where is the problem. Did you come across the situation? I will be appreciated with any help.

My hyperparameter setting:
    defaults = {"model": "exp1_model.npz",
                "attn-type": "deterministic", # Soft atten
                "dim-word": 512, # word vector dimensionality
                "ctx-dim": 512, # context vector dimensionality
                "dim": 1000, # the number of LSTM units
                "n-layers-att": 1, # number of layers used to compute the attention weights
                "n-layers-out": 1, # number of layers used to compute logit
                "n-layers-lstm": 1, # number of lstm layers
                "n-layers-init": 1, # number of layers to initialize LSTM at time 0
                "n-words": 5509, # vocab size
                "lstm-encoder": False, # if True, run bidirectional LSTM on input units
                "decay-c": 0., # weight decay coeff
                "alpha-c": 1., # doubly stochastic coeff
                "prev2out": True, # Feed previous word into logit
                "ctx2out": True, # Feed attention weighted ctx into logit
                "learning-rate": 0.01, # used only for SGD
                "optimizer": "adam",
                "selector": True, # selector (see paper)
                "use-dropout": True,  # setting this true turns on dropout at various points
                "use-dropout-lstm": False, # dropout on lstm gates
                "save-per-epoch": False, # this saves down the model every epoch
                "reload": False, 
              }

Note:"n-words": 5509, i get it from my training dataset. (5507words +2 ( <eos> and 'UNK'))

my input is like below:
(1) image feature:
image_conv_features1
image_conv_features2
...
image_conv_featuresn

(2) caption:
caption1 1
caption2 1
caption3 1
caption4 1
caption5 1
caption6 2
...

Note: caption1-5 belongs to the image 1, which is corresponding to the image_conv_features1

The wrong result I got:
...
Epoch 0 Update 46 Cost 142.9272 PD 0.26 UD 17.73
Epoch 0 Update 47 Cost 92.3364 PD 0.23 UD 7.20
Epoch 0 Update 48 Cost 133.7872 PD 0.26 UD 16.18
Epoch 0 Update 49 Cost 103.5006 PD 0.25 UD 9.56
Epoch 0 Update 50 Cost 112.8297 PD 0.25 UD 11.65
Saving...
Done
Truth  0 : 
people
hold
cardboard
frames
in
front
of
their
faces
Sample ( 0 )  0 : 
Truth  1 : 
picture
of
a
campsite
with
a
big
green
tent
Sample ( 0 )  1 : 
...

Problem1: Sample ( 0 )  0 -10: should have something be generated but it is nothing in here.
 
Also, with the epochs/updates increases, the validation will be like below:
Train  0 Valid  -0.00 Test  -0.00
